recipe/build.sh

```diff
@@ -1,15 +1,34 @@
 #!/bin/bash
-
-export CFLAGS="$CFLAGS $(gsl-config --cflags)"
-export LDFLAGS="$LDFLAGS $(gsl-config --libs)"
-
-# For whatever reason, it can't link to gsl correctly without this on OS X.
-export DYLD_FALLBACK_LIBRARY_PATH=$PREFIX/lib
```

Comment:
Did anyone test if these lines can be removed? They were included in the original PR to staged-recipes:

https://github.com/conda-forge/staged-recipes/pull/2783/files#diff-56af99ac0b5126b199f7f4d4be2a3a77

Reply:
Good catch. Saw it in the git diff of my update procedure ... still proceeded to see if CI will succeed. CI succeeds, but that is maybe not prove enough that everything is fine. Can also just keep those lines, they should be no harm.

Reply:
I don't understand those lines well enough to make the judgement call. It would be nice if someone could provide guidance on whether those lines make a difference in the ability of `r-gsl` to link to the correct version of GSL.

Reply:
Isn't there only one GSL present during the build?

Reply:
Looking through the build logs it appears to do the right thing.  The environment this is built in has basically nothing so it should not have another GSL present in a searchable path

Reply:
Cross-referencing also also https://github.com/conda-forge/r-rcppgsl-feedstock/pull/5 where the same lines got removed in the build.sh.